app/assets/src/components/views/PipelineViz.jsx

```diff
@@ -212,14 +326,14 @@ class PipelineViz extends React.Component {
       this.drawStageGraph(i);
     });
     this.adjustGraphNodePositions();
-    this.closeNonactiveSteps();
+    this.setUpGraphEventListeners();
   }
 
-  closeNonactiveSteps() {
-    this.stageNames.forEach((stageName, i) => {
+  setUpGraphEventListeners() {
```

Comment:
Not sure I fully understand this code... but seems to be inverted of what the communication should be.
Feels to me that part of this belongs on the js code, i.e. `NetworkGraph`.
Typically, you would pass an option `onClick` in the constructor just like a react prop.
Let me know if you want to discuss this offline.